### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by emmanuellmhango**
In this PR regarding the Blog app project, I have implemented the following requirements:

1. Created a method called current_user in the ApplicationController to make current user data available to all controllers. It returns the first user from the database.

2. Created forms to perform the following functions:
- To create a Post on behalf of the current_user.
- To create a comment on behalf of the current_user.

3. Implement functionality for users to add likes to the post.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
-    @posts = Post.includes(:author).where(author_id: params[:user_id])
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments)
   end
 
+  # def show
+  #  @user = User.find(params[:user_id])
+  #  @post = @user.posts.find(params[:id])
+  # end
+
```
Comment (Owusu-Desmond) [inline app/controllers/posts_controller.rb]: - [ ] Please make sure you remove commented code from your project to make your code clean. :pray: You should not push commented code to your repository. You can read more about it [here](https://www.freecodecamp.org/news/code-comments-the-good-the-bad-and-the-ugly-be9cc65fbf83/)

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
   def show
-    @post = Post.includes(:author).find(params[:id])
+    @post = Post.find(params[:id])
+    @comments = @post.comments.order(created_at: :desc)
+    render :show
   end
```
Comment (Owusu-Desmond) [inline app/controllers/posts_controller.rb]: - [ ] Please let's fix this issue :pray: After creating a post created by a user, _**I can't show see all posts and even show a specific post to test if one can comment and like**_. I am getting an error after creating a post and trying to show a specific post so make sure that you fix this error :blush:  You can see from the video below

  [Screencast from 03-10-2023 04:06:11 PM.webm](https://user-images.githubusercontent.com/92303891/224368626-01a17f5d-c8e6-4ee6-846c-680172f11b24.webm)

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @emmanuellmhango , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @emmanuellmhango, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**